src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -254,7 +255,13 @@ Object sanitizeObjectAndRecordMutation(@CheckForNull Object o, @CheckForNull Env
         if (modded != tempVal) {
             // Sanitization stripped out some values, so we need to record that and return modified version
             this.isUnmodifiedBySanitization = false;
-            return modded;
+            if (o instanceof UninstantiatedDescribable) {
+                // Need to retain the symbol.
```

Comment:
This is a fix to what is _stored_. Without this, even a fleshed-out version of `getResolvedArguments` would have no hope, because previously the storage would be something like

    delegate → [<anonymous> → 'TEST-*.xml']

with no record being kept that the original delegate was in fact `junit` ~ `JUnitResultArchiver`.